src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasCountrySlicer.java

```diff
@@ -0,0 +1,535 @@
+
+                // Record a successful slice
+                this.statistics.recordSlicedLine();
+            }
+            catch (final CoreException e)
+            {
+                // TODO - Consider shifting to a 4 digit namespace for identifiers
+                logger.error(
+                        "Country slicing exceeded maximum point identifier name space of {} for Line {}",
+                        AbstractIdentifierFactory.IDENTIFIER_SCALE, line.getIdentifier(), e);
+                this.statistics.recordSkippedLine();
+            }
+        }
+        else
+        {
+            // TODO - Consider expanding to a 4 digit namespace for identifiers
+            logger.error(
+                    "Country slicing exceeded maximum line identifier name space of {} for Line {}",
+                    AbstractIdentifierFactory.IDENTIFIER_SCALE, line.getIdentifier());
+            this.statistics.recordSkippedLine();
+        }
+    }
+
+    /**
+     * Converts the given {@link Line} to a JTS {@link Geometry}, slices the geometry and updates
+     * all corresponding {@link Point}s and {@link Line}s in the given raw Atlas.
+     *
+     * @param line
+     *            The {@link Line} to slice
+     */
+    private void sliceLine(final Line line)
+    {
+        this.statistics.recordProcessedLine();
+        final List<Geometry> slices = convertToJtsGeometryAndSlice(line);
+        processSlices(line, slices);
+    }
+
+    /**
+     * Slices all the {@link Line}s in the given raw Atlas.
+     */
+    private void sliceLines()
+    {
+        this.rawAtlas.lines().forEach(this::sliceLine);
+    }
+
+    /**
+     * Updates all points that haven't been assigned a country code after line-slicing. This
+     * includes any stand-alone points (e.g. trees, barriers) or points that fell outside of any
+     * country boundary.
+     */
+    private void slicePoints()
+    {
+        this.rawAtlas.points().forEach(point ->
+        {
+            final long pointIdentifier = point.getIdentifier();
+
+            // Only update points that haven't been assigned a country code after way slicing
+            if (!this.afterSlicingLinesAndPoints.getUpdatedPointTags().containsKey(pointIdentifier))
+            {
+                this.statistics.recordProcessedPoint();
+                this.afterSlicingLinesAndPoints.updatePointTags(pointIdentifier,
+                        createPointTags(point.getLocation(), true));
+            }
+        });
+    }
+
+    // TODO come back and verify we're keeping track of all required statistics
+
+    /**
+     * TODO
+     *
+     * @param relation
+     */
+    private void sliceRelation(final Relation relation)
+    {
+        this.statistics.recordProcessedRelation();
+
+        if (Validators.isOfType(relation, RelationTypeTag.class, RelationTypeTag.BOUNDARY,
+                RelationTypeTag.MULTIPOLYGON))
+        {
+            // Process geometry based relations
```

Comment:
This will come in another PR?

Reply:
Correct, coming in the next PR, currently in progress.